sdk/appconfiguration/Azure.Data.AppConfiguration/src/FeatureFlagConfigurationSetting.cs

```diff
@@ -0,0 +1,355 @@
+                foreach (var featureFlagFilter in _clientFilters)
+                {
+                    writer.WriteStartObject();
+                    writer.WriteString("name", featureFlagFilter.Name);
+                    writer.WritePropertyName("parameters");
+                    WriteParameterValue(writer, featureFlagFilter.Parameters);
+                    writer.WriteEndObject();
+                }
+                writer.WriteEndArray();
+            }
+            writer.WriteEndObject();
+            writer.WriteEndObject();
+            writer.Flush();
+
+            return Encoding.UTF8.GetString(memoryStream.ToArray());
+        }
+
+        private bool TryParseValue()
+        {
+            try
+            {
+                using var document = JsonDocument.Parse(_originalValue);
+                var root = document.RootElement;
+
+                if (!root.TryGetProperty("id", out var idProperty))
+                {
+                    return false;
+                }
+
+                _featureId = idProperty.GetString();
+
+                if (!root.TryGetProperty("enabled", out var enabledProperty))
+                {
+                    return false;
+                }
+
+                _isEnabled = enabledProperty.GetBoolean();
+
+                if (!root.TryGetProperty("conditions", out var conditionsProperty))
+                {
+                    return false;
+                }
+
+                if (root.TryGetProperty("description", out var descriptionProperty))
+                {
+                    _description = descriptionProperty.GetString();
+                }
+
+                if (root.TryGetProperty("display_name", out var displayNameProperty))
+                {
+                    _displayName = displayNameProperty.GetString();
+                }
+
+                if (conditionsProperty.TryGetProperty("client_filters", out var clientFiltersProperty))
+                {
+                    foreach (var clientFilter in clientFiltersProperty.EnumerateArray())
+                    {
+                        if (!clientFilter.TryGetProperty("name", out var filterNameProperty))
+                        {
+                            return false;
+                        }
+
+                        Dictionary<string, object> value = null;
+                        if (clientFilter.TryGetProperty("parameters", out var parametersProperty))
+                        {
+                            value = (Dictionary<string, object>)ReadParameterValue(parametersProperty);
+                        }
+
+                        _clientFilters.Add(new FeatureFlagFilter(filterNameProperty.GetString(), value ?? new Dictionary<string, object>()));
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static object ReadParameterValue(in JsonElement element)
```

Comment:
is there a reason why the serialization/deserialization happens here instead of the `ConfigurationServiceSerializer` class?

Reply:
Yes, this deserialization routine doesn't create a new `FeatureFlagConfigurationSetting` it modifies the existing instance. This is to support the case where somebody re-assigns the `Value` and we need to parse it again.


```
FeatureFlagConfigurationSetting setting = ...;

setting.Value = "{ \"featureId\":\"...\" ... }";